cms/cache/page.py

```diff
@@ -14,9 +14,14 @@
 from cms.toolbar.utils import get_toolbar_from_request
 from cms.utils import get_cms_setting
 from cms.utils.helpers import get_timezone_name
+from django.contrib.messages import get_messages
 
 
 def _page_cache_key(request):
+    # do not store nor retrieve page in cache if some messages are going to be displayed
+    messages = get_messages(request)
+    if len(messages) or getattr(messages, 'used', False) or getattr(messages, 'added_new', False):
```

Comment:
I would like to avoid bundling the check logic to this function.
This function should always generate the key regardless of conditions.

I would create a function called "can_cache_page" or something like that which encapsulates the logic in [here](https://github.com/divio/django-cms/blob/release/3.4.x/cms/cache/page.py#L38) as well as the messages check.